how-pinterest-reduced-costs-and-improved-data-consistency-with-newsql-database.md

```diff
@@ -0,0 +1,171 @@
+---
+title: How Pinterest Reduced Costs and Improved Data Consistency with a NewSQL Database
+author: ['Lianghong Xu']
+date: 2021-09-08
+summary: Pinterest used the HBase ecosystem for their data storage. But this infrastructure was complex, with limited functionalities and high maintenance costs. They switched to TiDB over YugaByteDB and CockroachDB because it best met their requirements for stability and performance.
+tags: ['Scalability']
+customer: Pinterest
+customerCategory: Internet
+logo: /images/blog/customers/pinterest-logo.png
+image: /images/blog/pinterest-reduced-costs-and-improved-data-consistency-with-newsql-database.jpg
+---
+
+**Industry:** Media and Entertainment
+
+**Author:** Lianghong Xu (Engineering Manager for Storage & Cache Team at Pinterest)
+
+**Editors:** [Yajing Wang](https://github.com/anotherrachel), [Caitin Chen](https://github.com/anotherrachel), Tom Dewan
+
+![Pinterest Migrated from HBase to a NewSQL Database](media/pinterest-reduced-costs-and-improved-data-consistency-with-newsql-database.jpg)
+
+[Pinterest](https://en.wikipedia.org/wiki/Pinterest) is an American visual discovery engine for finding ideas like recipes, home and style inspiration, and more. As of March 2021, it had more than 478 million global monthly active users.
+
+In recent years, we used the HBase ecosystem to serve our critical online applications. But this **infrastructure was complex**. It had **limited functionalities** and brought **high maintenance and infrastructure costs**. We looked for a new storage solution that supports **distributed transactions** and **SQL compatibility**.
+
+After we compared YugaByteDB, CockroachDB, and [TiDB](https://docs.pingcap.com/tidb/stable), an open-source scale-out database, we chose TiDB because it best met our requirements for stability and performance. TiDB helps us **reduce system complexity and infrastructure cost and achieve stronger consistency**.
+
+In this article, I'll describe:
+
+* [The storage evolution at Pinterest](#evolution-of-pinterest-online-storage)
+* [Our HBase ecosystem](#hbase-at-pinterest)
+* [Our exploration of distributed SQL](#in-search-of-new-storage)
+* [TiDB's challenges and opportunities](#tidb-challenges-and-opportunities)
+
+## Evolution of Pinterest online storage
+
+* In 2012, we introduced sharded MySQL into Pinterest's tech stack. We used it to store the core data models including pins, boards, and users.
+* In 2013, we introduced HBase as Pinterest’s first NoSQL database system. Since then, HBase has been very popular, and it has been serving many of our critical services. Based on HBase, we’ve built a columnar data store, a graph service, and a number of other applications.
```

Comment:
```suggestion
* In 2013, we introduced HBase as Pinterest's first NoSQL database system. Since then, HBase has been very popular, and it has been serving many of our critical services. Based on HBase, we’ve built a columnar data store, a graph service, and a number of other applications.
```